pkg/services/cloudmigration/api/api.go

```diff
@@ -315,14 +318,38 @@ func (cma *CloudMigrationAPI) CreateSnapshot(c *contextmodel.ReqContext) respons
 	defer span.End()
 
 	uid := web.Params(c.Req)[":uid"]
-
 	if err := util.ValidateUID(uid); err != nil {
 		span.SetStatus(codes.Error, "invalid session uid")
 		span.RecordError(err)
 
 		return response.ErrOrFallback(http.StatusBadRequest, "invalid session uid", err)
 	}
 
+	var cmd CreateSnapshotRequestDTO
+	if err := web.Bind(c.Req, &cmd); err != nil {
+		span.SetStatus(codes.Error, "invalid request body")
+		span.RecordError(err)
+
+		return response.ErrOrFallback(http.StatusBadRequest, "invalid request body", err)
+	}
+
+	if len(cmd.ResourceTypes) == 0 {
+		return response.ErrOrFallback(http.StatusBadRequest, "at least one resource type is required", cloudmigration.ErrEmptyResourceTypes)
+	}
+
+	rawResourceTypes := make([]cloudmigration.MigrateDataType, 0, len(cmd.ResourceTypes))
+	for _, t := range cmd.ResourceTypes {
+		rawResourceTypes = append(rawResourceTypes, cloudmigration.MigrateDataType(t))
+	}
+
+	_, err := cloudmigration.ResourceDependency.Parse(rawResourceTypes)
```

Comment:
Can you please add more documentation to the HTTP method description about what a valid payload looks like?

Reply:
This will change but yes!